LONDON | May-2025 | Anuar Duisenbek | Module-Structuring-and-Testing-Data | Sprint-1

AnuarBey opened the pull request:
## Learners, PR Template

Self checklist

- [x] I have committed my files one by one, on purpose, and for a reason
- [x] I have titled my PR with REGION | COHORT_NAME | FIRST_NAME LAST_NAME | PROJ_NAME
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/contributing/)
- [x] My changes meet the [requirements](./README.md) of this task

## Changelist

 Resolved all errors in the Errors section: ran files with Node.js... interpreted the error messages, and fixed the bugs using error documentation

 Interpreted and analyzed larger programs in the Interpret section: used console.log and MDN to understand syntax, operators, and functions

cjyuan commented on Sprint-1/1-key-exercises/4-random.js:
```diff
@@ -7,3 +7,9 @@ const num = Math.floor(Math.random() * (maximum - minimum + 1)) + minimum;
 // Try logging the value of num and running the program several times to build an idea of what the program is doing
+
+/* This code generates a random whole number between 1 and 100.
+ Math.random() returns a decimal between 0 and 1 (not including 1).
```
You can also use the concise and precise ***interval notation*** to describe a range of values.
For example, we can say, "`Math.random()` returns a random number in the interval `[0, 1)`"

cjyuan commented on Sprint-1/3-mandatory-interpret/3-to-pounds.js:
```diff
@@ -23,5 +23,37 @@ console.log(`£${pounds}.${pence}`);
+
+Extract the pound part (everything except the last 2 characters)
+
+5.const pence = paddedPenceNumberString
+  .substring(paddedPenceNumberString.length - 2)
+  .padEnd(2, "0");
+  Extract the pence part (last 2 characters), and make sure it's 2 digits long
```
Could we expect this program to work as intended for any valid `penceString` if we deleted `.padEnd(2, "0")` from the code?
In other words, do we really need `.padEnd(2, "0")` in this script?

cjyuan reviewed (commented):
You could consider adding `.idea` into `.gitignore` so that the folder won't get pushed to GitHub. The files in that folder don't appear to be necessary for the exercises.

Explanation is clear . 

I only have one question for your regarding usage of `.padEnd()`.

AnuarBey commented:
Thank you for pointing out the .padEnd() usage. After reviewing the code, I realised it isn’t really necessary here, as the output is correct for all valid inputs even without it. I’ve removed .padEnd() as suggested — thanks for the reminder!

Also, I have now added .idea/ to .gitignore so it won’t be pushed to the repository in the future.
